espaloma/data/qcarchive_utils.py

```diff
@@ -21,35 +21,69 @@
 # =============================================================================
 # UTILITY FUNCTIONS
 # =============================================================================
-def get_client():
-    return ptl.FractalClient()
+def get_client(url: str = "api.qcarchive.molssi.org") -> qcportal.client.PortalClient:
+    """
+    Returns a instance of the qcportal client.
+
+    Parameters
+    ----------
+    url: str, default="api.qcarchive.molssi.org"
+        qcportal instance to connect
+
+    Returns
+    -------
+    qcportal.client.PortalClient
+        qcportal client instance.
+    """
+    # Note, this may need to be modified to include username/password for non-public servers
+    return qcportal.PortalClient(url)
 
 
 def get_collection(
-    client,
-    collection_type="OptimizationDataset",
-    name="OpenFF Full Optimization Benchmark 1",
+        client,
+        collection_type="optimization",
+        name="OpenFF Full Optimization Benchmark 1",
 ):
-    collection = client.get_collection(
-        collection_type,
-        name,
+    """
+    Connects to a specific dataset on qcportal
+
+    Parameters
+    ----------
+    client: qcportal.client, required
+        The qcportal client instance
+    collection_type: str, default="optimization"
+        The type of qcarchive collection, options are
+        "torsiondrive", "optimization", "gridoptimization", "reaction", "singlepoint" "manybody"
+    name: str, default="OpenFF Full Optimization Benchmark 1"
+        Name of the dataset
+
+    Returns
+    -------
+    (qcportal dataset, list(str))
+        Tuple with an instance of qcportal dataset and list of record names
+
+    """
+    collection = client.get_dataset(
+        dataset_type=collection_type,
+        dataset_name=name,
     )
 
-    record_names = list(collection.data.records)
+    record_names = collection.entry_names
 
     return collection, record_names
 
 
-def get_graph(collection, record_name):
-    # get record and trajectory
-    record = collection.get_record(record_name, specification="default")
-    entry = collection.get_entry(record_name)
+def process_record(record, entry):
+    """
+    Processes a given record/entry pair from a dataset and returns the graph
+    """
+
     from openff.toolkit.topology import Molecule
 
-    mol = Molecule.from_qcschema(entry)
+    mol = Molecule.from_qcschema(entry.dict())
 
     try:
-        trajectory = record.get_trajectory()
+        trajectory = record.trajectory
     except:
```

Comment:
I know you didn't add this in the PR, but do we know what kind of error we are catching here? I've wasted a lot of debugging time before with naked `except`. If we don't have any idea and it isn't easy to figure out, doing something like `except Exception as e:` and then we can at least `print(e)` and have some hope of figuring it out in the future.

Reply:
I'll stick a todo note in here regarding checking the logic.  This may have been necessary depending on how the old version handled cases where a computation failed before it produced any data (maybe the function `get_trajectory()` would just fail instead of producing an empty list?).  A second quest is, this only works for optimization datasets that have trajectories @yuanqing-wang @kntkb  would it be useful to change around the code here to also to support single point energy calculations (i.e., single point energy calculations don't have a trajectory field defined, would need to just be from geometry)